app/models/llms/basic_workflow/cli.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
+import argparse
+import sys
+from typing import Optional
+
+from app.config.logging_config import get_logger
+from .integration import ItineraryGenerator, save_to_session_history, save_conversation_to_file
+
+logger = get_logger("voyager_t800.cli")
+
+
+class VoyagerCLI:
+    def __init__(self, args):
+        self.generator = ItineraryGenerator()
+        self.session_history = []
+        self.args = args
+    
+    def _save_and_exit(self):
+        if self.session_history:
+            print("💾 Saving conversation before exit...")
+            save_conversation_to_file(self.session_history)
+        print("👋 Thank you for using Voyager T800!")
+
+    def _exit_without_saving(self):
+        print("👋 Thank you for using Voyager T800!")
+
+    def _exit(self):
+        if not self.args.not_save_history:
```

Comment:
The flag `not_save_history` is negative and may be confusing. Prefer a positively-named flag (e.g., `save_history`) for clarity, unless this is required for backward compatibility.